### ISSUE TITLE
Multi-class in `ConfusionMetrixMetric`

### ISSUE THREAD

**Issue by KumoLiu** [Project-MONAI/MONAI#7009]
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/7002

<div type='discussions-op-text'>

<sup>Originally posted by **mcleod-matthew-gene** September 19, 2023</sup>
Hello All,

For the MONAI [ConfusionMatrixMetric](https://docs.monai.io/en/stable/metrics.html#monai.metrics.ConfusionMatrixMetric), how does one have it calculate the multi-class metric rather than multilabel metric. I don't see a parameter to pass in to tell it do multi-class metrics and none of the reduction methods seem to yield multiclass results, rather they are multilabel. 

For example, see below. `y` and `y_preds` are of shapes `(4,3)`, so batch size of 4 and 3 classes. This should be a simple multiclass problem, but as the code snippet highlights, the results returned by MONAI are correct if this is a multilabel problem. 

<img width="692" alt="Screen Shot 2023-09-15 at 10 58 29 AM" src="https://github.com/Project-MONAI/MONAI/assets/133425655/cd4e4c7d-d49d-400c-a108-08f88d5b7212">

The documentation of the ConfusionMatrix states:  `It can support both multi-classes and multi-labels classification and segmentation tasks`. How do I get multi-class metrics?

Thanks!
</div>

**Comment by KumoLiu** [comment]
Looks like a limitation in our [ConfusionMatrixMetric](https://docs.monai.io/en/stable/metrics.html#monai.metrics.ConfusionMatrixMetric), now we only support one-hot format data. For multi-class classifications, such as the example above:
```
from monai.metrics import ConfusionMatrixMetric
a = torch.tensor([[1, 0, 0], [0, 1, 0], [1, 0, 0], [0, 0, 1]])
b = torch.tensor([[1, 0, 0], [0, 1, 0], [0, 0, 1], [0, 0, 1]])

cm = ConfusionMatrixMetric(metric_name="acc", reduction="none")
cm(a, b)
cm.aggregate()
--> tensor([[1., 1., 1.],
        [1., 1., 1.],
        [0., 1., 0.],
        [1., 1., 1.]])

cm = ConfusionMatrixMetric(metric_name="acc", reduction="mean_channel")
cm(a, b)
cm.aggregate()
--> [tensor([1.0000, 1.0000, 0.3333, 1.0000])]
# which will consider class 1 as TN for the third sample.
```

I'm not sure if this is what we expected; if not, perhaps we should enhance the class or at the least update the docstring.
What do you think?
cc @wyli @ericspod @yiheng-wang-nv

**Comment by mcleod-matthew-gene** [comment]
Any resolution to this thread? The documentation clearly states that multiclass `ConfusionMatrixMetrics` are supported, but seems to not work.

**Comment by vikashg** [comment]
closing as alternatives exist @mcleod-matthew-gene please reopen if you still need help.
